x-pack/plugins/security_solution/public/timelines/store/timeline/helpers.ts

```diff
@@ -1349,3 +1353,26 @@ export const updateFilters = ({ id, filters, timelineById }: UpdateFiltersParams
     },
   };
 };
+
+interface UpdateExcludedRowRenderersIds {
+  id: string;
+  excludedRowRendererIds: RowRendererId[];
+  timelineById: TimelineById;
+}
+
+export const updateExcludedRowRenderersIds = ({
+  id,
+  excludedRowRendererIds,
+  timelineById,
+}: UpdateExcludedRowRenderersIds): TimelineById => {
+  const timeline = timelineById[id];
+
+  return {
+    ...timelineById,
+    [id]: {
+      ...timeline,
+      excludedRowRendererIds,
+      showRowRenderers: excludedRowRendererIds.length !== Object.keys(RowRendererId).length,
```

Comment:
`showRowRenderers` feels like "derived" state that may not be required on the model itself. consider removing it